src/View/Antlers/Language/Runtime/NodeProcessor.php

```diff
@@ -1538,10 +1542,20 @@ public function reduce($processNodes)
 
                             $sectionName = 'section:'.$tagMethod.'__yield'.$activeYield;
 
+                            $sections = null;
+
+                            if ($this->cascade == null) {
+                                // If our current cascade is null (most likely from deep isolation/etc.
+                                // attempt to locate the sections from a global Cascade instance.
+                                $sections = app(Cascade::class)->sections();
+                            } else {
+                                $sections = $this->cascade->sections();
+                            }
+
                             LiteralReplacementManager::registerRegionReplacement(
                                 $sectionName,
                                 $tagMethod,
-                                $this->cascade->sections()->get($tagMethod)
+                                $sections->get($tagMethod)
```

Comment:
If I revert this and run the whole suite, the only thing that fails is `Antlers/Runtime/YieldTest@test_normal_yield`. But if I run that test by itself, it passes.

This tells me that there's some static variable being set in another test that effects this one.

Also, if you remove this entire if/else and just replace it with `$sections = app(Cascade::class)->sections()`, the whole suite passes. This sorta defeats the purpose of passing around the cascade though, so I'd rather not do it. 🤔 

Any idea as to what could be sticking around between tests?

If we're able to figure that out, then reverting this chunk would be the better code, IMO.

Reply:
Figured it out - need to have an exception triggered in a previous tag callback - this will cause the correct setting to not be reapplied after its done. Updated all places that would set the `$requiresRuntimeIsolation` flag to wrapped in a `try catch finally` block. Restoration of the correct value will happen in the `finally` block, but the exception is simply thrown afterwards